src/map/skill.cpp

```diff
@@ -8725,7 +8724,7 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 			if (sd && sd->sc.data[SC_PETROLOGY_OPTION])
 				brate = sd->sc.data[SC_PETROLOGY_OPTION]->val3;
 
-			if (tsc && tsc->data[type]) {
+			if (tsc && tsc->data[SC_STONE]) {
 				status_change_end(bl, type, INVALID_TIMER);
```

Comment:
Before this would mean it cancels SC_STONE (being Stone+Stonewait).
Do we need to check and remove both now?

Reply:
Can confirm that on official servers "Stonecurse" removes both status changes (Stone and Stonewait). This is the case for both pre-re and re.